src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala

```diff
@@ -0,0 +1,72 @@
+package com.kinja.amqp
+
+import com.kinja.amqp.model.Message
+import com.kinja.amqp.model.MessageConfirmation
+
+import org.slf4j.{ Logger => Slf4jLogger }
+
+import scala.util.Failure
+import scala.util.Success
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object UnconfirmedMessageRepeater {
+
+	/**
+	 * Loads the messages from the store.
+	 * For already confirmed messages, deletes the message and the confirmation.
+	 * For unconfirmed messages, tries to republish and if succeeds, deletes old message and confirmation.
+	 */
+	def resendUnconfirmedMessages(messageStore: MessageStore, olderThan: Long, producer: AmqpProducer, logger: Slf4jLogger): Unit = {
+		val oldMessages = messageStore.loadMessageOlderThan(olderThan)
```

Comment:
yeah, absolutely. in an ideal situation i would actually do 1 at a time, to minimize the risk of resending the same message as many times as the number of application servers that run this code